src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -614,7 +614,8 @@ public void UpdatePackageSource(PackageSource source, bool updateCredentials, bo
             if (string.Equals(newSource.Name, existingSource.Name, StringComparison.OrdinalIgnoreCase))
             {
                 if ((!string.Equals(newSource.Source, existingSource.Source, StringComparison.OrdinalIgnoreCase) ||
-                    newSource.ProtocolVersion != existingSource.ProtocolVersion) && newSource.IsPersistable)
+                    newSource.ProtocolVersion != existingSource.ProtocolVersion ||
```

Comment:
cc @Nigusu-Allehu 

I'm guessing DisableTLSCertificateValidation has the same concern?
Can you please double check?

Reply:
https://github.com/NuGet/Home/issues/13423